keras_cv/layers/vit_class_tokenizing.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class ClassTokenizing(layers.Layer):
+    """
+    Layer to concat learnable class token to input for Vision Transformers from:
```

Comment:
Not sure which part of this layer is "learnable"?

Reply:
Changed the class token to be initialized as a learnable parameter in the new push. I have to test whether it compiles right now 🤔

Reply:
Yeah, it's throwing an exception because of the unexpected size. Changed to accept the input shape since it's internally being used only with patches that are computed at the time, so we can guarantee input shapes. Let me verify if this works